src/main/java/org/dsa/iot/alarm/AlarmUtil.java

```diff
@@ -0,0 +1,191 @@
+                new Value(normalTime),
+                new Value(ackTime),
+                new Value(record.getAckUser()),
+                new Value(record.getMessage()),
+                new Value(record.getHasNotes())
+        ));
+    }
+
+    /**
+     * Encodes the columns for an action that returns a table of alarms.
+     */
+    public static void encodeAlarmColumns(Action action) {
+        action.addResult(new Parameter("UUID_STR",ValueType.STRING));
+        action.addResult(new Parameter(SOURCE_PATH,ValueType.STRING));
+        action.addResult(new Parameter("Alarm Class",ValueType.STRING));
+        action.addResult(new Parameter("Created Time",ValueType.STRING));
+        action.addResult(new Parameter("Created State", ENUM_ALARM_TYPE));
+        action.addResult(new Parameter("Normal Time",ValueType.STRING));
+        action.addResult(new Parameter("Ack Time",ValueType.STRING));
+        action.addResult(new Parameter("Ack User",ValueType.STRING));
+        action.addResult(new Parameter(MESSAGE,ValueType.STRING));
+        action.addResult(new Parameter("Has Notes",ValueType.BOOL));
+    }
+
+    /**
+     * Enqueues the parameter into the alarming thread pool.
+     */
+    public static void enqueue(Callable callable) {
+        Objects.getDaemonThreadPool().submit(callable);
+    }
+
+    /**
+     * Enqueues the parameter into the alarming thread pool.
+     */
+    public static void enqueue(Runnable runnable) {
+        Objects.getDaemonThreadPool().submit(runnable);
+    }
+
+    /**
+     * Creates a daemon thread to execute the given runnable.
+     * @param runnable What to run.
+     * @param threadName Optional, name to give the thread.
+     */
+    public static void run(Runnable runnable, String threadName) {
+        Thread thread = new Thread(runnable);
+        if (threadName == null) {
+            threadName = "Alarm Link";
+        }
+        thread.setName(threadName);
+        thread.setDaemon(true);
+        thread.start();
+    }
+
+    /**
+     * Ensures that a runtime exception is thrown. If the given error is not a
+     * runtime exception, then one will thrown with the parameter as the inner
+     * exception.
+     */
+    public static void throwRuntime(Throwable error) {
+        if (error instanceof RuntimeException) {
+            throw (RuntimeException) error;
+        }
+        throw new RuntimeException(error);
+    }
+
+    /**
+     * Looks for the JAVA_TYPE config and if found, instantiates an instance and calls
+     * init for the given node thus, loading the entire subtree of AlarmObjects.
+     * @param node The node the alarm object represents/proxies.
+     * @return An instance of the javaType config, or null if there isn't one.
+     * @throws RuntimeException Which will wrap the true exception if there are
+     * issues instantiating the type.
+     */
+    public static AlarmObject tryCreateAlarmObject(Node node) {
```

Comment:
I don't think I like this method returning `null`s. This adds lots of null checks in the code everywhere.